server/src/test/java/org/elasticsearch/rest/RestControllerTests.java

```diff
@@ -623,6 +623,39 @@ public void testDispatchRestrictSystemIndices() {
         assertFalse(context.isSystemIndexAccessAllowed());
     }
 
+    public void testDispatchCompatibleHandler() {
+        final String mimeType = randomFrom("application/vnd.elasticsearch+json;compatible-with=7");
+        String content = randomAlphaOfLength((int) Math.round(BREAKER_LIMIT.getBytes() / inFlightRequestsBreaker.getOverhead()));
+        final List<String> contentTypeHeader = Collections.singletonList(mimeType);
+        FakeRestRequest fakeRestRequest = new FakeRestRequest.Builder(NamedXContentRegistry.EMPTY)
+            .withContent(new BytesArray(content), RestRequest.parseContentType(contentTypeHeader)).withPath("/foo")
+            .withHeaders(Map.of("Content-Type", contentTypeHeader, "Accept", contentTypeHeader))
```

Comment:
Here we are passing in both Content-Type and Accept headers. Should we test one or the other or both ? (if so a random selection would be good)

Reply:
as per https://github.com/elastic/elasticsearch/pull/53228#discussion_r391180225
I think we should assume both should be present.
WDYT? 
I will then reply with the conclusion on https://github.com/elastic/elasticsearch/issues/52370

Reply:
Let's punt on it for now, the details there are important, but not immediately. I added a "parking lot" section on https://github.com/elastic/elasticsearch/issues/51816 so we don't forget.